### PR TITLE
Fix el7 detection

### LINKED ISSUE #437: ubi7/python-38 uses new django-ex -> failing test
When test suite is ran for `rh-python38-container`:
```
[FAILED] for  test_from_dockerfile
...
django.core.exceptions.ImproperlyConfigured: SQLite 3.8.3 or later is required (found 3.7.17).
```

### PR DESCRIPTION
**Pull request by frenzymadness**
Fixes #437

### CODE REVIEW

**Comment by frenzymadness** [comment]
[test][test-openshift]

**Comment by pkubatrh** [comment]
LGTM

**Comment by phracek** [comment]
[test]

**Comment by phracek** [comment]
CentOS7 failed because of #412 

```bash
15:29:39 [1m[INFO] Running test test_scl_variables_in_dockerfile ... [0m
15:29:39 
15:29:39 
15:29:39 [1m[INFO] Testing variable presence during `docker exec`[0m
15:29:39 
15:29:41  Value /opt/rh/rh-nodejs14/root/usr/share/man is missing from variable MANPATH
15:29:41 MANPATH=/opt/rh/rh-python27/root/usr/share/man:/opt/rh/python27/root/usr/share/man:/opt/rh/httpd24/root/usr/share/man:/opt/rh/rh-nodejs10/root/usr/share/man
15:29:41 
```

**Comment by phracek** [comment]
[test][test-openshift]

**Review by phracek — APPROVED** [review]
LGTM.

**Comment by frenzymadness** [comment]
[test][test-openshift]

**Comment by phracek** [comment]
RHEL7 test failed with reason https://gist.github.com/rhscl-automation/f25fb03415b8bc596d63dfce8962f474#file-s2i-python-container-113-sh-L15010

```bash

[1m[INFO] Test from Dockerfile[0m
test/run: line 223: podman: command not found
Using this Dockerfile:
FROM rhscl/python-38-rhel7:1
```

File: 2.7/test/run [hunk 1]
```diff
@@ -219,7 +219,8 @@ test_from_dockerfile(){
   # and no longer supports Python 2. So on all CentOS/RHEL 7 images and all Python 2 images,
   # we're using the old Django LTS version 1.11.x from the main git branch of the django-ex project.
   # In all other cases we're using the newer Django LTS version 2.2.x from the corresponding git branch.
-  if [[ ${VERSION} == "2.7" ]] || [[ ${IMAGE_NAME} =~ "centos7" ]] || [[ ${IMAGE_NAME} =~ "rhel7" ]] || [[ ${IMAGE_NAME} =~ "ubi7" ]]; then
+
+  if [[ ${VERSION} == "2.7" ]] || podman inspect ${IMAGE_NAME} --format "{{.Config.Env}}" | tr " " "\n" | grep -q "^PLATFORM=el7"; then
```
Comment (phracek) [inline 2.7/test/run:223]: I would prefer to use `docker` and not `podman`

File: 3.6/test/run [hunk 2]
```diff
@@ -219,7 +219,8 @@ test_from_dockerfile(){
   # and no longer supports Python 2. So on all CentOS/RHEL 7 images and all Python 2 images,
   # we're using the old Django LTS version 1.11.x from the main git branch of the django-ex project.
   # In all other cases we're using the newer Django LTS version 2.2.x from the corresponding git branch.
-  if [[ ${VERSION} == "2.7" ]] || [[ ${IMAGE_NAME} =~ "centos7" ]] || [[ ${IMAGE_NAME} =~ "rhel7" ]] || [[ ${IMAGE_NAME} =~ "ubi7" ]]; then
+
+  if [[ ${VERSION} == "2.7" ]] || podman inspect ${IMAGE_NAME} --format "{{.Config.Env}}" | tr " " "\n" | grep -q "^PLATFORM=el7"; then
```
Comment (phracek) [inline 3.6/test/run:223]: The same here.

File: 3.8/test/run [hunk 3]
```diff
@@ -219,7 +219,8 @@ test_from_dockerfile(){
   # and no longer supports Python 2. So on all CentOS/RHEL 7 images and all Python 2 images,
   # we're using the old Django LTS version 1.11.x from the main git branch of the django-ex project.
   # In all other cases we're using the newer Django LTS version 2.2.x from the corresponding git branch.
-  if [[ ${VERSION} == "2.7" ]] || [[ ${IMAGE_NAME} =~ "centos7" ]] || [[ ${IMAGE_NAME} =~ "rhel7" ]] || [[ ${IMAGE_NAME} =~ "ubi7" ]]; then
+
+  if [[ ${VERSION} == "2.7" ]] || podman inspect ${IMAGE_NAME} --format "{{.Config.Env}}" | tr " " "\n" | grep -q "^PLATFORM=el7"; then
```
Comment (phracek) [inline 3.8/test/run:223]: The same here

File: 3.9/test/run [hunk 4]
```diff
@@ -219,7 +219,8 @@ test_from_dockerfile(){
   # and no longer supports Python 2. So on all CentOS/RHEL 7 images and all Python 2 images,
   # we're using the old Django LTS version 1.11.x from the main git branch of the django-ex project.
   # In all other cases we're using the newer Django LTS version 2.2.x from the corresponding git branch.
-  if [[ ${VERSION} == "2.7" ]] || [[ ${IMAGE_NAME} =~ "centos7" ]] || [[ ${IMAGE_NAME} =~ "rhel7" ]] || [[ ${IMAGE_NAME} =~ "ubi7" ]]; then
+
+  if [[ ${VERSION} == "2.7" ]] || podman inspect ${IMAGE_NAME} --format "{{.Config.Env}}" | tr " " "\n" | grep -q "^PLATFORM=el7"; then
```
Comment (phracek) [inline 3.9/test/run:223]: The same here

File: test/run [hunk 5]
```diff
@@ -220,7 +220,9 @@ test_from_dockerfile(){
   # and no longer supports Python 2. So on all CentOS/RHEL 7 images and all Python 2 images,
   # we're using the old Django LTS version 1.11.x from the main git branch of the django-ex project.
   # In all other cases we're using the newer Django LTS version 2.2.x from the corresponding git branch.
-  if [[ ${VERSION} == "2.7" ]] || [[ ${IMAGE_NAME} =~ "centos7" ]] || [[ ${IMAGE_NAME} =~ "rhel7" ]] || [[ ${IMAGE_NAME} =~ "ubi7" ]]; then
+{% raw %}
+  if [[ ${VERSION} == "2.7" ]] || podman inspect ${IMAGE_NAME} --format "{{.Config.Env}}" | tr " " "\n" | grep -q "^PLATFORM=el7"; then
```
Comment (phracek) [inline test/run:224]: Replace `podman` with `docker` and generate new files and tests should passed ;)

**Review by phracek — CHANGES_REQUESTED** [review]

**Comment by frenzymadness** [comment]
Thanks @phracek !

[test][test-openshift]

**Comment by phracek** [comment]
OpenShift tests failed, because of OpenShift does not start up.

Let's try again.

[test-openshift]

**Review by phracek — APPROVED** [review]
LGTM

**Pull request merged**